internal/pkg/cli/pipeline_deploy.go

```diff
@@ -446,13 +516,17 @@ func (o *deployPipelineOpts) deployPipeline(in *deploy.CreatePipelineInput) erro
 	}
 
 	// If the stack already exists - we update it
-	shouldUpdate, err := o.shouldUpdate()
-	if err != nil {
-		return err
-	}
-	if !shouldUpdate {
-		return nil
+
+	if !o.showDiff {
+		shouldUpdate, err := o.shouldUpdate()
+		if err != nil {
+			return err
+		}
+		if !shouldUpdate {
+			return nil
+		}
```

Comment:
chatted with @huanjani about this offline, but i wonder if we should just remove this logic entirely (the `shouldUpdate()` stuff, which you didn't add! just adding to this pr since it came up here 😛)? it feels kinda odd to ask "do you want to deploy" after someone typed `pipeline deploy` 🤔

Reply:
would be fine to do as a separate PR if you'd prefer :)

Reply:
I can make a separate PR for this so that it makes clear that I am just making this change